### DISCUSSION TITLE
Anyone using mysql2 npm package with the latest app dir beta?

### DISCUSSION THREAD

**Discussion by Nithur-M** [vercel/next.js#46810] [Help]
### Summary

I am facing an issue. I think it is caused by the module `lru-cache` used my mysql2 package. 
I don't know how to resolve it or get rid of it. Please help. Thanks.

### Additional information

```js
`TypeError: LRU is not a constructor`
at `node_modules\mysql2\lib\parsers\parser_cache.js (5:20)`
```

**Comment by CanRau** [comment]
Same problem here, interestingly only when I try to use my db in `middleware.ts`.
No problem in routes and pages 🤷🏼 

Though really need to use it in `middleware.ts` as well 😅 

@Nithur-M did you find a solution/workaround by any chance?
